### PR TITLE
chore(RHINENG-7042): adopt fec-config

### PR DESCRIPTION
**Pull request by mkholjuraev**
# Description

Associated Jira ticket: # (issue)

Resolves: https://issues.redhat.com/browse/RHINENG-7042

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

File: fec.config.js
Comment (bastilian) [inline fec.config.js:13]: I'm very suspicious of this. Do we really need to add this to every app, even if the react router version is v6?

I think we only need to do this if the version is not v6, like policies for example.

Comment (bastilian) [reply fec.config.js:13]: I mean both `expose` and `shared`.

Comment (mkholjuraev) [reply fec.config.js:13]: Not all apps have upgraded to v6. Can this be a reason to have this in place still?

Comment (mkholjuraev) [reply fec.config.js:13]: I understand that once all are migrated, we were expected to remove this.

Comment (bastilian) [reply fec.config.js:13]: @mkholjuraev I really think we only need to do this in apps that are not migrated, and not in all till all are migrated. Chrome is sharing its version, which is v6. This would cause issues in apps with v5 and there we can add these definitions to make the app share its version instead of chrome.

Comment (bastilian) [reply fec.config.js:13]: If without this the app still works proper, we probably don't need it.

Comment (mkholjuraev) [reply fec.config.js:13]: I have tried it out. I have removed both exclude&shared configs related to router, and the UI does not work with this error: **useRoutes() may be used only in the context of a <Router> component.**

Comment (mkholjuraev) [reply fec.config.js:13]: Call Stack:
 invariant
  vendors-node_modules_patternfly_react-component-groups_node_modules_clsx_dist_clsx_mjs-node_m-af6625.js:12071:11
 useRoutesImpl
  vendors-node_modules_patternfly_react-component-groups_node_modules_clsx_dist_clsx_mjs-node_m-af6625.js:34809:102
 useRoutes
  vendors-node_modules_patternfly_react-component-groups_node_modules_clsx_dist_clsx_mjs-node_m-af6625.js:34804:10
 Routes
  vendors-node_modules_patternfly_react-component-groups_node_modules_clsx_dist_clsx_mjs-node_m-af6625.js:35693:10
 renderWithHooks
  chrome-root.b25b7d405e94b287.js:37887:18
 mountIndeterminateComponent
  chrome-root.b25b7d405e94b287.js:41651:13
 beginWork
  chrome-root.b25b7d405e94b287.js:43164:16
 HTMLUnknownElement.callCallback
  chrome-root.b25b7d405e94b287.js:25746:14
 Object.invokeGuardedCallbackDev
  chrome-root.b25b7d405e94b287.js:25795:16
 invokeGuardedCallback
  chrome-root.b25b7d405e94b287.js:25859:31

File: fec.config.js
Comment (bastilian) [inline fec.config.js:19]: I am wondering what the best practice is with these version props and what would be the best to avoid running into issues. 

I'm thinking maybe they should be a bit more open. Similar to the sem ver definitions: https://github.com/RedHatInsights/frontend-components/blob/master/packages/components/package.json#L45-L46

For the `requiredVersion` we could use `>=6.0.0 <7.0.0` and I think we can drop the `version`, as webpack will default to the version in `package.json`

File: fec.config.js
Comment (bastilian) [inline fec.config.js:8]: This can be omitted as `interceptChromeConfig` is checked to be explicitly set to `true`.

File: fec.config.js
Comment (mkholjuraev) [reply fec.config.js:19]: is not `requiredVersion: '^6.8.1' `equal to `requiredVersion: ' >=6.0.0 <7.0.0'`?

Comment (bastilian) [reply fec.config.js:19]: I'm not sure we need it to be exactly higher than `6.8.1`, or?

In general unless we have a very specific reason to stick to a very specific version, we should keep them "unspecific". 6.8.1 is not even the version that is in the lock file, it is `6.22.3`, or in the package.json, which is `^6.11.1`,

This leads to confusion, which is the correct version. Let's stick to `>=X.0.0` definitions. Then we can also just copy and paste these things, as they'll apply equally in each project.

File: fec.config.js
Comment (mkholjuraev) [reply fec.config.js:17]: Looking at the documentation, I believe the default value is not false for the import prop rather it is the name of the provided package. I tried removing it and I got the same error as above ^. Maybe I do not comprehend the complexity of this shared scope logic.

Comment (bastilian) [reply fec.config.js:17]: Ah, my bad. I thought that first part below the prop title in the docs states the default.

File: fec.config.js
Comment (mkholjuraev) [reply fec.config.js]: @bastilian I have made the necessary changes, PTAL

File: fec.config.js
Comment (bastilian) [inline fec.config.js:17]: The `version` will default to the `package.json` version, so this can be removed.

Comment (bastilian) [reply fec.config.js:17]: Oh, right, Ok. Then let's leave it for now, but I'll investigate why we have this guard in the frontend components federated modules module.

Comment (mkholjuraev) [reply fec.config.js:17]: Will merge once, there is an approval

**Review by bastilian — APPROVED** [review]
Let's get this merged! Thank you @mkholjuraev!

**Pull request merged**
